parsers/julia.c

```diff
@@ -1148,36 +1205,46 @@ static void parseModule (lexerState *lexer, vString *scope, int parent_kind)
 /* Import format:
  * [ "import" | "using" ] <ident> [: <name>]
  */
-static void parseImport (lexerState *lexer, vString *scope, int parent_kind)
+static void parseImport (lexerState *lexer, vString *scope, int parent_kind, int token_type)
 {
     vString *name = vStringNew();
-
+    int library_role;
+    int unknown_role;
+    if (token_type == TOKEN_IMPORT)
+    {
+        library_role = JULIA_LIBRARY_IMPORTED;
+        unknown_role = JULIA_UNKNOWN_IMPORTED;
+    }
+    else /* if (token_type) == TOKEN_USING */
+    {
+        library_role = JULIA_LIBRARY_USING;
+        unknown_role = JULIA_UNKNOWN_USING;
+    }
     /* capture the imported name */
     advanceToken(lexer, true);
+    vStringCopy(name, lexer->token_str);
```

Comment:
`.` may be included in vStringValue (lexer->token_str);

So...
```
char * dot = strchr (vString(lexer->token_str_, '.');
if (dot)
{
     vString *library_part = vStringNewNInit(vString(lexer->token_str), dot -  vString(lexer->token_str));
     vString *unknown_part = vStringNewInit(dot + 1);
     /* make_ref_tag for library_part */
    /* make_ref_tag for unknown_part */
    /* delete the strings */
}
else
....

Reply:
Thanks for the tip! I think I can finish this without touching the lexer now.